pkg/reconciler/v1alpha1/clusteringress/config/istio.go

```diff
@@ -31,26 +33,51 @@ const (
 	// IngressGatewayKey is the name of the configuration entry
 	// that specifies ingress gateway url.
 	IngressGatewayKey = "ingress-gateway"
+
+	// GatewayKeyPrefix is the prefix of all keys to configure Istio gateways.
+	GatewayKeyPrefix = "gateway."
 )
 
+// IngressGateway specifies the name of the Gateway and the K8s Service backing it.
+type IngressGateway struct {
+	GatewayName string
+	ServiceURL  string
+}
+
 // Istio contains istio related configuration defined in the
 // istio config map.
 type Istio struct {
 	// IngressGateway specifies the ingress gateway url.
-	IngressGateway string
+	IngressGateways []IngressGateway
 }
 
 // NewIstioFromConfigMap creates an Istio config from the supplied ConfigMap
 func NewIstioFromConfigMap(configMap *corev1.ConfigMap) (*Istio, error) {
-	gateway, ok := configMap.Data[IngressGatewayKey]
-	if !ok {
-		return nil, fmt.Errorf("failed to fetch %s from configmap %s", IngressGatewayKey, IstioConfigName)
+	urls := map[string]string{}
+	gatewayNames := []string{}
+	for k, v := range configMap.Data {
+		if strings.HasPrefix(k, GatewayKeyPrefix) {
```

Comment:
nit: `if !... { continue }` to reduce indent.

Reply:
Done